snap/info.go

```diff
@@ -173,3 +173,25 @@ type AppInfo struct {
 func (app *AppInfo) SecurityTag() string {
 	return fmt.Sprintf("snap.%s.%s", app.Snap.Name(), app.Name)
 }
+
+// WrapperPath returns the path to wrapper invoking the app binary.
+func (app *AppInfo) WrapperPath() string {
+	var binName string
+	if app.Name == app.Snap.Name() {
+		binName = filepath.Base(app.Name)
+	} else {
+		binName = fmt.Sprintf("%s.%s", app.Snap.Name(), filepath.Base(app.Name))
+	}
+
+	return filepath.Join(dirs.SnapBinariesDir, binName)
+}
+
+// ServiceFile returns the systemd service file path for the daemon app.
+func (app *AppInfo) ServiceFile() string {
+	return filepath.Join(dirs.SnapServicesDir, fmt.Sprintf("snap_%s_%s_%d.service", app.Snap.Name(), app.Name, app.Snap.Revision))
```

Comment:
The prefix seems necessary to avoid conflicts with similar content (there is, for example, `"dbus-org.bluez.service"`, which would conflict with a snap `"dbus-org"` with app `"bluez"`).

That said, can we please use `app.SecurityTag()+".service"` here? This is the same logic we've used for apparmor, seccomp, etc, and it ends up as `"snap.<snap>.<app>.service"`. So it both looks nice and creates a feeling of commonality that people can start to expect.